Documentation/WritingDocForExtension/Migrate.rst

```diff
@@ -32,32 +32,222 @@ a migration tool. This tool can be used to automatically create a
+-----------------------------------
+
+After you have created the :file:`guides.xml` file, you can remove the old
+:file:`Settings.cfg` file.
+
+You can also delete the files :file:`genindex.rst` which was previously used
+to generate an index.
+
+The :file:`screenshots.json` is also no longer evaluated, so you can remove
+that optional file.
+
+3. Adapt :file:`Includes.rst.txt`
+---------------------------------
+
+The main documentation directory can contain a file :file:`Includes.rst.txt`
+to include any fixed text, which will be placed on every page of your rendered
+documentation.
+
+Previously it was also used to define a list of utilized directives/roles.
+
+You can either safely remove that file, or add your fixed text to it.
+
+Most official documentation uses this as the stub of the file:
+
+..  code-block:: text
+    :caption: /Documentation/Includes.rst.txt
+
+    ..  You can put central messages to display on all pages here
+
+4. Remove the entry `genindex` from :file:`Index.rst` (Index/Glossary)
+----------------------------------------------------------------------
+
+If you previously had a :file:`genindex.rst` file, this optional index
+(or glossary) was rendered as a page through an entry in the file :file:`Index.rst`
+like this:
+
+..  code-block:: text
+    :caption: /Documentation/Index.rst
+    :emphasize-lines: 17
+
+    **Table of Contents:**
+
+    .. toctree::
+       :maxdepth: 2
+       :titlesonly:
+
+       Introduction/Index
+       Installation/Index
+       Details/Index
+
+    .. Meta Menu
+
+    .. toctree::
+       :hidden:
+
+       Sitemap
+       genindex
+
+Remove the entry `genindex` from the list.
+
+..  hint::
+
+    See :ref:`migrate-glossary` for details about the future of the
+    index (glossary) generation.
+
+5. Prevent code-snippets with :file:`.rst` extension
+----------------------------------------------------
+
+All files ending in :file:`.rst` will be interpreted by the new rendering,
+and every file that is just a code snippet placed in an external file
+should be renamed to use a :file:`.rst.txt` extension instead.
+
+Recommendations
+===============
+
+The following list is not a requirement to utilize the new rendering, but
+follows "best practice" to make the most of your documentation project.
+
+..  _migrate-to-makefile:
+
+Add a :file:`Makefile` to your project
+--------------------------------------
+
+A :file:`Makefile` is a simple command line runner configuration file, which requires
+the utility `GNU make <https://www.gnu.org/software/make/manual/make.html>`_
+to be available on your Operating System.
+
+This allows you to perform shortcuts to render your documentation instead
+of typing a long :bash:`docker run...` or :bash:`podman run...` command:
+
+..  code-block:: shell
+
+    make docs
+
+For inspiration, check out the :file:`Makefile` of the main PHP-based rendering
+repository:
+
+https://github.com/TYPO3-Documentation/render-guides/blob/main/Makefile
```

Comment:
This is too big and the most tasks are not necessary and might confuse users. I suggest to remove this reference.

Reply:
Hm, I do think this is a very nice example of a complex but useful Makefile, and I explicitly said "for inspiration"? I'd like to keep this.

Reply:
I would agree to keep it, it is just for inspiration

Reply:
This is about docs rendering not what you can do with a Makefile.

Reply:
This is simply an inspiration what else you can do with the tool